torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -2755,7 +2755,23 @@ RegisterOperators reg2({
 
     DEFINE_BINARY_OP(aten::add, a + b),
     DEFINE_BINARY_OP(aten::sub, a - b),
-    DEFINE_BINARY_OP(aten::mul, a * b),
+    DEFINE_BINARY_OP(aten::mul, a* b),
+
+    // int ** int produces a float, because negative exponents produce float
```

Comment:
This does not match with the python semantics:

```
>>> a = 2
>>> type(a)
<class 'int'>
>>> c = a ** -1
>>> type(c)
<class 'float'>
>>> b = a ** 2
>>> type(b)
<class 'int'>
```
Python still returns type int for positive exponents. This should ideally be an overloaded return type in the schema so that we could base on the value of the second arg to return the correct type.

Reply:
```
 2 ** -2
0.25
```

Positive exponents isn't a type here - the type needs to be defined statically. An alternative is to return scalar here but i think that's worse.

Reply:
This does not fix too much on the problem because we were always returning int before, now we are always returning float, I would imagine the return type `int` case is more frequent than the `float` case. This ideally should be some return type like `Union[int, float]` but given that we don't want to do that I guess we probably have to choose one.

Reply:
It preserves more of the information, because you can just cast the output to `int()` instead of always returning `0` in the cast of a negative output.